### PR TITLE
Avoid temporary allocations

### PR DESCRIPTION
**Pull request by uklotzde**
...and many refactorings for type-safe (newtype + destructuring) and more idiomatic Rust code.

Decoupling the file path from the track avoids to store redundant data that is no needed.

### CODE REVIEW

File: src/tag/container.rs [hunk 1]
```diff
@@ -282,15 +282,15 @@ impl TagContainer {
                         continue;
                     }
                     markers::MarkerType::Cue => {
-                        if marker.start_position_millis == None {
+                        if marker.start_position == None {
```
Comment (Holzhaus) [inline src/tag/container.rs:285]: Hmm, why did you rename this? Isn't it better to have the unit as part of the name, so it's clear that these are not samples/frames/seconds/etc.?

Comment (uklotzde) [reply src/tag/container.rs:285]: Replaced by a type-safe and more expressive newtype. Just have a look at the type definition ;)

Comment (uklotzde) [reply src/tag/container.rs:285]: Well, not "newtype" but a struct that correctly *names* the value.

**Review by Holzhaus — APPROVED** [review]
